### PR TITLE
feat: Auto Recharge Credits + Credit Packs

### PR DESCRIPTION
**Pull request by nickscamara**
see firecrawl-web#112 for change explanation.

### CODE REVIEW

**Review by rafaelsideguide — APPROVED** [review]

File: apps/api/src/controllers/auth.ts [hunk 1]
```diff
@@ -75,23 +75,27 @@ export async function setCachedACUC(
 
 export async function getACUC(
   api_key: string,
-  cacheOnly = false
+  cacheOnly = false,
+  useCache = true
```
Comment (mogery) [inline apps/api/src/controllers/auth.ts:79]: Not caching ACUC is getting rid of like, 90% of the benefit of even having ACUC.

**Review by mogery — APPROVED** [review]

File: apps/api/src/controllers/auth.ts (hunk 1 above)
Comment (nickscamara) [reply apps/api/src/controllers/auth.ts:79]: This is only used once, to prevent race conditions on charging.

File: apps/api/src/services/billing/stripe.ts
Comment (nickscamara) [inline apps/api/src/services/billing/stripe.ts:1]: This probably needs quite a bit of work to get it right, will iterate after.

**Review by nickscamara — COMMENTED** [review]
merging

**Pull request merged**